core/src/main/java/io/ably/lib/realtime/ConnectionRecoveryKey.java

```diff
@@ -0,0 +1,62 @@
+package io.ably.lib.realtime;
+
+import com.google.gson.JsonSyntaxException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.ably.lib.util.Log;
+import io.ably.lib.util.Serialisation;
+
+public class ConnectionRecoveryKey {
+    private static final String TAG = "RecoveryKey";
+
+    private final String connectionKey;
+    private final long msgSerial;
+    /**
+     * Key - channel name
+     * <p>
+     * Value - channelSerial
+     */
+    private final Map<String, String> serials = new HashMap<>();
+
+    public ConnectionRecoveryKey(String connectionKey, long msgSerial) {
+        this.connectionKey = connectionKey;
+        this.msgSerial = msgSerial;
+    }
+
+    public String getConnectionKey() {
+        return connectionKey;
+    }
+
+    public long getMsgSerial() {
+        return msgSerial;
+    }
+
+    public Map<String, String> getSerials() {
+        return serials;
+    }
+
+    public void setSerials(Map<String, String> serials) {
+        this.serials.clear();
+        this.serials.putAll(serials);
+    }
+
+    public void addSerials(String channelName, String channelSerial) {
```

Comment:
Since you are only adding a single serial at a time, shouldn`t this method be called `addSerial`? 🤔 
```suggestion
    public void addSerial(String channelName, String channelSerial) {
```